### PR TITLE
HTTPException displays description when raised, fixing #728

### PR DESCRIPTION
**Pull request by shellhead**
Changed the string representation of `HTTPException` to include the description, for easier debugging (as requested in #728).

The `code` attribute was also causing `__str__` to blow up, since it's default value is `None` and it was expecting a digit in the formatting.

### CODE REVIEW

File: tests/test_exceptions.py [hunk 1]
```diff
@@ -70,13 +70,20 @@ def test_aborter_custom():
...
+        'spelling and try again.')
     assert repr(exc) == "<NotFound '404: Not Found'>"
 
     exc = exceptions.NotFound('Not There')
-    assert text_type(exc) == '404: Not Found'
+    assert text_type(exc) == '404 Not Found: Not There'
     assert repr(exc) == "<NotFound '404: Not Found'>"
 
+    exc = exceptions.HTTPException('An error message')
+    assert text_type(exc) == 'None Unknown Error: An error message'
```
Comment (untitaker) [inline tests/test_exceptions.py]: This isn't particularly pretty, could we replace `None` with something like `???` (or omit it entirely)?

**Comment by untitaker** [comment]
Perfect except for the one nitpick.

**Comment by shellhead** [comment]
Thanks for the feedback. I have a busy start of the weekend, but should be able to look at it on Sunday.

**Comment by shellhead** [comment]
I went with `???`, since it provides some sort of hint to the user that `Exception.code` hasn't been set. Maybe something a bit more descriptive, like `NO_HTTP_STATUS_CODE` would be better? Changed the `__repr__` to follow the same behavior.

**Comment by untitaker** [comment]
Thanks!

**Pull request merged**